### ISSUE TITLE
Resolving relative and direct paths

### ISSUE THREAD

**Issue by soupman99** [wallabyjs/public#2015]
### Issue description or question
I'm sure there are already 10 examples of this but I've search and didn't find them. 

I can't seem to get any paths to resolve who live outside of ./src/app .  How would I get these paths to resolve with both typescript & wallaby?

 Here's a simple example built by extending ngCliWebpackSample.  
```
import { Component } from '@angular/core'; 
import * as app from "../../package.json"; //<-- Module not found: Error: Can't resolve '../../package.json' in './src/app
import { environment } from 'src/environments/environment';  //<-- Module not found: Error: Can't resolve 'src/environments/environment' in './src/app

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',    
  styleUrls: ['./app.component.css'],
})
export class AppComponent {
  title = 'app';
  constructor(){
    console.log('Version', app.version)
    console.log('Environment', environment.production)
  }
}
```

### Tsconfig in root of project
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "module": "es2015",
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "resolveJsonModule": true,
    "importHelpers": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2018",
      "dom"
    ]
  },
  "exclude": ["src/wallabyTest.ts"]
}

```

### Wallaby.js configuration file

```javascript
var wallabyWebpack = require('wallaby-webpack');
var path = require('path');

var compilerOptions = Object.assign(
  require('./tsconfig.json').compilerOptions,
  require('./src/tsconfig.spec.json').compilerOptions);

module.exports = function (wallaby) {

  var webpackPostprocessor = wallabyWebpack({
    entryPatterns: [
      'src/wallabyTest.js',
      'src/**/*spec.js'
    ],

    module: {
      rules: [
        {test: /\.css$/, loader: ['raw-loader']},
        {test: /\.html$/, loader: 'raw-loader'},
        {test: /\.ts$/, loader: '@ngtools/webpack', include: /node_modules/, query: {tsConfigPath: 'tsconfig.json'}},
        {test: /\.js$/, loader: 'angular2-template-loader', exclude: /node_modules/},
        {test: /\.styl$/, loaders: ['raw-loader', 'stylus-loader']},
        {test: /\.less$/, loaders: ['raw-loader', {loader: 'less-loader'}]},
        {test: /\.scss$|\.sass$/, loaders: ['raw-loader', 'sass-loader']},
        {test: /\.(jpg|png|svg)$/, loader: 'raw-loader'}
      ]
    },

    resolve: {
      extensions: ['.js', '.ts'],
      modules: [
        path.join(wallaby.projectCacheDir, 'src/app'),
        path.join(wallaby.projectCacheDir, 'src'),
        
        'node_modules'
      ]
    },
    node: {
      fs: 'empty',
      net: 'empty',
      tls: 'empty',
      dns: 'empty'
    }
  });

  return {
    files: [
      {pattern: 'src/**/*.+(ts|css|less|scss|sass|styl|html|json|svg)', load: false},
      {pattern: 'src/**/*.d.ts', ignore: true},
      {pattern: 'src/**/*spec.ts', ignore: true}
    ],

    tests: [
      {pattern: 'src/**/*spec.ts', load: false},
      {pattern: 'src/**/*e2e-spec.ts', ignore: true}
    ],

    testFramework: 'jasmine',

    compilers: {
      '**/*.ts': wallaby.compilers.typeScript(compilerOptions)
    },

    middleware: function (app, express) {
      var path = require('path');
      app.use('/favicon.ico', express.static(path.join(__dirname, 'src/favicon.ico')));
      app.use('/assets', express.static(path.join(__dirname, 'src/assets')));
    },

    env: {
      kind: 'chrome'
    },

    postprocessor: webpackPostprocessor,

    setup: function () {
      window.__moduleBundler.loadTests();
    },

    debug: true
  };
};
```

### Code editor or IDE name and version

VScode 1.31.1 

### OS name and version
OSX 10.14.3

**Comment by ArtemGovorov** [comment]
Just 2 simple changes in your Wallaby config are required:

> import * as app from "../../package.json"; //<-- Module not found: Error: Can't resolve '../../package.json' in './src/app

All files used in your app [need to be in the `files` list](https://wallabyjs.com/docs/config/files.html):
```diff
...
    files: [
      {pattern: 'src/**/*.+(ts|css|less|scss|sass|styl|html|json|svg)', load: false},
      {pattern: 'src/**/*.d.ts', ignore: true},
      {pattern: 'src/**/*spec.ts', ignore: true}
+     {pattern: 'package.json', load: false, instrument: false},
    ],
```

> import { environment } from 'src/environments/environment';  //<-- Module not found: Error: Can't resolve 'src/environments/environment' in './src/app

With the way you import the module (as `src/environments/environment` as opposed to `environments/environment`), you need to add the project's root into Webpack `resolve.modules` setting:

```diff
...
    resolve: {
      extensions: ['.js', '.ts'],
      modules: [
        path.join(wallaby.projectCacheDir, 'src/app'),
        path.join(wallaby.projectCacheDir, 'src'),
+       wallaby.projectCacheDir,
        'node_modules'
      ]
    },
```

**Comment by soupman99** [comment]
It partially worked. Now I”m getting:

Uncaught TypeError: __webpack_require__.t is not a function.


I’ve tested and narrowed it down to being caused by package.json not
src/environment/environment


On March 1, 2019 at 9:06:50 AM, Artem Govorov (notifications@github.com)
wrote:

Just 2 simple changes in your Wallaby config are required:

import * as app from "../../package.json"; //<-- Module not found: Error:
Can't resolve '../../package.json' in './src/app

All files used in your app need to be in the files list
<https://wallabyjs.com/docs/config/files.html>:

...
    files: [
      {pattern:
'src/**/*.+(ts|css|less|scss|sass|styl|html|json|svg)', load: false},
      {pattern: 'src/**/*.d.ts', ignore: true},
      {pattern: 'src/**/*spec.ts', ignore: true}+     {pattern:
'package.json', load: false, instrument: false},
    ],

import { environment } from 'src/environments/environment'; //<-- Module
not found: Error: Can't resolve 'src/environments/environment' in './src/app

With the way you import the module (as src/environments/environment as
opposed to environments/environment), you need to add the project's root
into Webpack resolve.modules setting:

...
    resolve: {
      extensions: ['.js', '.ts'],
      modules: [
        path.join(wallaby.projectCacheDir, 'src/app'),
        path.join(wallaby.projectCacheDir, 'src'),+
wallaby.projectCacheDir,
        'node_modules'
      ]
    },

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/wallabyjs/public/issues/2015#issuecomment-468675659>,
or mute the thread

**Comment by smcenlly** [comment]
Can you please provide us with a sample repo with the same problem? We'd be happy to take a look but without that can't think of what might be causing your problem.

**Comment by soupman99** [comment]
@smcenlly Thank you so much for your help!

Added here:
https://github.com/soupman99/ng-cli-webpack-sample

**Comment by ArtemGovorov** [comment]
Thanks for providing the repo! 

The issue was with `wallaby-webpack` package and it is fixed now in v3.9.14. Please [update to the latest version](https://github.com/soupman99/ng-cli-webpack-sample/pull/1) of the package and it should work for you.
